src/assets/RestApi/rest-api.md

```diff
@@ -0,0 +1,400 @@
+**- Resource Representation:** The data returned by an endpoint is typically in a structured format, such as JSON or XML, which represents the resource or resources associated with the endpoint.
+
+## Implementing a REST API
+Now, let's start to build a REST API. Before we start, make sure you have Python installed on your system. You can download Python from [python.org](https://www.python.org/downloads). We will also need Flask, which you can install using pip:
+```bash
+pip install Flask
+```
+
+We will build a REST API for managing a list of tasks. Each task will have a title and a status indicating whether it's completed or not.
+
+1. Create a new Python file: `app.py`
+
+2. Import Flask and initialize the app:
+```python
+from flask import Flask, jsonify, request
+
+app = Flask(__name__)
+```
+
+3. Define the data structure for our tasks. For simplicity, we'll use a list of dictionaries:
+```python
+tasks = [
+    {"id": 1, "title": "Buy groceries", "completed": False},
+    {"id": 2, "title": "Clean the house", "completed": True}
+]
+```
+
+### GET Requests
+A GET request is used to retrieve data from a server. We can get all the avaliable data or a specific data using a GET request.
+- To retrieve all tasks:
+```python
+@app.route("/tasks", methods=["GET"])
+def get_tasks():
+    return jsonify({"tasks": tasks}), 200
+```
+- The `@app.route` decorator provided by Flask binds a URL pattern to the function that follows it. In this case, the URL pattern in `/tasks`.
+- `methods=["GET]` specifies that this route should respond to the HTTP GET requests. This means that when a client sends a GET request to the `/tasks` URL, Flask will call the associated function.
+- `get_tasks()` function will return a JSON response containing all tasks.
+- `jsonify({"tasks": tasks})` is a Flask function that converts the Python dictionary `{"tasks": tasks}` into a JSON response.
+
+- To retrieve a specific task:
+```python
+@app.route("/tasks/<int:task_id>", methods=["GET"])
+def get_task(task_id):
+    task = next((item for item in tasks if item["id"] == task_id), None)
+    if task is None:
+        return jsonify({"error": "Task not found"}), 404
+    return jsonify({"task": task}), 200
+```
+- `@app.route` binds a URL pattern `/tasks/<int:task_id>` to the function that follows it.
+- `<int:task_id>` is a variable part of the URL that will be captured and passed to the function as an argument.
+- `methods=["GET"]` specifies that this route should respond to the HTTP GET requests. This means that when a client sends a GET request to the `tasks/<int:task_id>` URL, Flask will call the associated function.
+- `get_task()` function will take `task_id` as argument that is extracted from `/tasks/<int:task_id>` URL. The function will then search for a task with that ID in the `tasks` list. If a task is found, it will return the task in JSON format. If no task is found, it will return a `404` status code with an error message.
```

Comment:
```suggestion
- `get_task()` function will take `task_id` as the argument that is extracted from `/tasks/<int:task_id>` URL. The function will then search for a task with that ID in the `tasks` list. If a task is found, it will return the task in JSON format. If no task is found, it will return a `404` status code with an error message.
```